src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotificationsTrait.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.plugins.bitbucket;
+
+import com.cloudbees.jenkins.plugins.bitbucket.hooks.WebhookConfiguration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import jenkins.scm.api.SCMSource;
+import jenkins.scm.api.trait.SCMSourceContext;
+import jenkins.scm.api.trait.SCMSourceTrait;
+import jenkins.scm.api.trait.SCMSourceTraitDescriptor;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * A {@link SCMSourceTrait} for {@link BitbucketSCMSource} that sets the committersToIgnore
+ * setting in {@link WebhookConfiguration}.
+ *
+ * @since 2.10.0
+ */
+public class BitbucketBuildStatusNotificationsTrait extends SCMSourceTrait {
+
+    /**
+     * Should unstable builds be communicated as success to Bitbucket
+     */
+    @NonNull
+    private final boolean isUnstableBuildSuccess;
+
+    /**
+     * Constructor.
+     *
+     * @param isUnstableBuildSuccess should unstable builds be communicated as success to Bitbucket
+     */
+    @DataBoundConstructor
+    public BitbucketBuildStatusNotificationsTrait(@NonNull boolean isUnstableBuildSuccess) {
+        this.isUnstableBuildSuccess = isUnstableBuildSuccess;
+    }
+
+    /**
+     * @return if unstable builds will be communicated as successful
+     */
+    public boolean isUnstableSuccess() {
+        return this.isUnstableBuildSuccess;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void decorateContext(SCMSourceContext<?, ?> context) {
+        ((BitbucketSCMSourceContext) context).withIsUnstableBuildSuccess(isUnstableSuccess());
+    }
+
+    /**
+     * Our constructor.
+     */
+    @Symbol("bitbucketBuildStatusConfiguration")
```

Comment:
```suggestion
    @Symbol("bitbucketNotifications")
```

Reply:
What do these symbols do?

Reply:
@ssc327 They control the name of this trait in pipeline and JobDSL.

Reply:
Got it. So if I want to make this more generic, should I rename the class and update the symbol?

Reply:
Actually, are there other types if Bitbucket notifications beyond build notifications? Otherwise is not the current name general enough?

Reply:
How about this?
```suggestion
    @Symbol("bitbucketBuildStatusNotifications")
```

Reply:
I like it.